bigquery/google/cloud/bigquery/query.py

```diff
@@ -351,7 +351,7 @@ def run(self, client=None):
         :param client: the client to use.  If not passed, falls back to the
                        ``client`` stored on the current dataset.
         """
-        if self._job is not None:
+        if self.job is not None:
```

Comment:
`job` is a property which lazily initializes `_job` from the server-supplied properties.